### PR TITLE
{Misc} Remove direct call to `msrestazure`

### PR DESCRIPTION
**Pull request by evelyn-ys**
**Related command**

**Description**

We are planning to drop `msrestazure` dependency as described in it's disclaimer

> This package is deprecated and no longer receives updates
> 
> The authentication part of this package has been moved to [azure-identity](https://pypi.org/project/azure-identity/)
> 
> The other parts of this library are covered by [azure-mgmt-core](https://pypi.org/project/azure-mgmt-core/)
> 
> As such, we will no longer accept PR and fix issues on this project.

So this PR mainly works on migrating
- functions in `msrestazure.tools` to `azure.mgmt.core.tools`
- `msrestazure.azure_exceptions.CloudError` to `azure.core.exceptions.HttpResponseError`

There're some other `msrestazure` calls left including
- vendored track1 sdk in `dla` module (will deprecate in next breaking change window #29879 )
- vendored track1 sdk in `dls` module (cc @akharit for awareness)
- vendored track1 data plane sdk in `keyvault` module (will discuss with keyvault service team)
- `msrestazure.azure_cloud._populate_from_metadata_endpoint` call in `cloud` module (cc @jiasli for awareness)
- `msrestazure.azure_operation.AzureOperationPoller` in `resource` module (cc @zhoxing-ms for awareness)

We will handle these cases in separate PRs

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Misc

**Review by FumingZhang — APPROVED** [review]
Approve for the change related to `acs` (AKS, `src/azure-cli/azure/cli/command_modules/acs`). Thanks!

**Review by ambrahma — APPROVED** [review]
`RDBMS` has reviewed this PR.

**Comment by yonzhan** [comment]
Please fix conflicting files

File: src/azure-cli/azure/cli/command_modules/synapse/manual/operations/sqlpoolsensitivitylabel.py [hunk 1]
```diff
@@ -69,8 +69,8 @@ def sqlpool_sensitivity_label_update(
         sensitivity_label.information_type = current_label.information_type
         sensitivity_label.information_type_id = current_label.information_type_id
 
-    except CloudError as ex:
-        if not (ex.error and ex.error.error and 'SensitivityLabelsLabelNotFound' in ex.error.error):
+    except HttpResponseError as ex:
+        if not (ex.status_code and ex.status_code == '404'):
```
Comment (wanyang7) [inline src/azure-cli/azure/cli/command_modules/synapse/manual/operations/sqlpoolsensitivitylabel.py:73]: if not (ex and 'SensitivityLabelsLabelNotFound' in str(ex)):

Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/synapse/manual/operations/sqlpoolsensitivitylabel.py]: updated as request, pls review again

**Comment by jiasli** [comment]
This is a very big RP. I would suggest creating separate PRs for each change:

1. Migrate `parse_resource_id`, `resource_id`, `is_valid_resource_id` from `msrestazure.tools` to `azure.mgmt.core.tools`
2. Migrate `msrestazure.azure_exceptions.CloudError` to `azure.core.exceptions.HttpResponseError`

**Review by calvinhzy — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Review by wanyang7 — APPROVED** [review]

**Comment by wanyang7** [comment]
Approve for the change related to `synapse`

**Comment by jiasli** [comment]
I actually did the `msrestazure.tools` to `azure.mgmt.core.tools` migration for Azure CLI Core in https://github.com/Azure/azure-cli/pull/20740.

**Review by jiasli — APPROVED** [review]

**Comment by jiasli** [comment]
I am actually considering moving `parse_resource_id`, `is_valid_resource_id`, `resource_id` to CLI Core (https://github.com/Azure/azure-cli/pull/29973), so that command modules take no direct dependency on `azure.cli.core.arm_tools`. If there is a Track 3 SDK in the future, the migration would also be easier.

**Review by yonzhan — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

**Comment by evelyn-ys** [comment]
`msrestazure.azure_exceptions.CloudError` to `azure.core.exceptions.HttpResponseError` missed one in https://github.com/Azure/azure-cli/blob/05a5d8fa878d8ae4d2ee8a1294feedc8af83fedb/src/azure-cli-core/azure/cli/core/util.py#L258-L262

Will fix in separate PR

**Pull request merged**